src/backend/joanie/core/models/products.py

```diff
@@ -845,12 +845,23 @@ def enroll_user_to_course_run(self):
                     course_run_id=open_course_run_id, user=self.owner
                 )
             except Enrollment.DoesNotExist:
-                Enrollment.objects.create(
+                enrollment = Enrollment.objects.create(
                     course_run_id=open_course_run_id,
                     is_active=True,
                     user=self.owner,
                     was_created_by_order=True,
                 )
+                if (
+                    enrollment.is_active
+                    and enrollment.user.enrollments.exclude(id=enrollment.id)
+                    .filter(
+                        course_run__course=enrollment.course_run.course,
+                        course_run__end__gte=now,
+                        is_active=True,
+                    )
+                    .exists()
+                ):
+                    enrollment.delete()
```

Comment:
I don't understand what you did here. Furthermore, an enrollment should never be deleted just disactivated.

Reply:
It was related to a test that verifies the flow of the order update method. Precisely, it was this [line](https://github.com/openfun/joanie/blob/1288503f0fb1df3a1de3cecbd0803f44d2170a02/src/backend/joanie/core/flows/order.py#L408).
Without adding this section here, I would end up having 2 enrollments instead of 1 according to this test : 
`test_flows_order_validate_auto_enroll_failure`

Furthermore, why it should delete the latest created enrollment in this if statement, this is the reason : 
```
            message = _(
                "You are already enrolled to an opened course run "
                f'for the course "{self.course_run.course.title}".'
            )
```

Reply:
Let's discuss about that tomorrow.